nodeadm/cmd/nodeadm/init/init.go

```diff
@@ -67,6 +67,12 @@ func (c *initCmd) Run(log *zap.Logger, opts *cli.GlobalOptions) error {
 	}
 	log.Info("Loaded configuration", zap.Reflect("config", nodeConfig))
 
+	// Set environment variables early so all AWS SDK calls and system processes use them
+	log.Info("Configuring system environment variables..")
+	if err := handleSystemEnvironmentVariables(log, nodeConfig.Spec.Instance); err != nil {
```

Comment:
optional: could be nice to just pass in nodeConfig.Spec.Instance.environment since you don't use the localStorage field that I can see.